### ISSUE TITLE
no function clause matching in Mongo.Protocol.handle_info/2 ... :ssl_closed

### ISSUE THREAD

**Issue by alextitze** [elixir-mongo/mongodb#169]
Hi,

getting the following error from time to time:

```
Child DBConnection.Poolboy.Worker of Supervisor #PID<0.15357.0> (:poolboy_sup) terminated
** (exit) an exception was raised:
    ** (FunctionClauseError) no function clause matching in Mongo.Protocol.handle_info/2
        (mongodb) lib/mongo/protocol.ex:132: Mongo.Protocol.handle_info({:ssl_closed, {:sslsocket, {:gen_tcp, #Port<0.29392>, :tls_connection, :undefined}, #PID<0.15373.0>}}, %{connection_type: :client, database: "db", host: "example.com", request_id: 190, socket: {:ssl, {:sslsocket, {:gen_tcp, #Port<0.29392>, :tls_connection, :undefined}, #PID<0.15373.0>}}, ssl: true, timeout: 5000, topology_pid: #PID<0.15325.0>, wire_version: 2, write_concern: %{w: 1}})
        (db_connection) lib/db_connection/connection.ex:558: DBConnection.Connection.do_handle_info/2
        (connection) lib/connection.ex:810: Connection.handle_async/3
        (stdlib) gen_server.erl:616: :gen_server.try_dispatch/4
        (stdlib) gen_server.erl:686: :gen_server.handle_msg/6
        (stdlib) proc_lib.erl:247: :proc_lib.init_p_do_apply/3
Pid: #PID<0.15365.0>
Start Call: DBConnection.Poolboy.Worker.start_link/?
```

Should this be added to `lib/mongo/protocol.ex` next to the `def handle_info({:tcp_closed, _}, s)` function?
```
def handle_info({:ssl_closed, _}, s) do
  err = Mongo.Error.exception(tag: :ssl, action: "async recv", reason: :closed)
  {:disconnect, err, s}
end
```

Thanks!

**Comment by ankhers** [comment]
Thanks for finding this. That looks like it should do the trick. Do you mind sending a PR?

**Comment by alextitze** [comment]
No problem, will do!

**Comment by alextitze** [comment]
https://github.com/ankhers/mongodb/pull/170

**Comment by ankhers** [comment]
Fixed in #170
